app/models/financial_year.rb

```diff
@@ -449,6 +436,64 @@ def split_into_periods(interval)
     end
   end
 
+  def all_previous_financial_years_closed_or_locked?
+    years = FinancialYear.where("started_on < ?", self.started_on).order(:started_on)
+    years.all? { |y| y.locked? || y.closed? }
+  end
+
+  def no_draft_entry?
+    journal_entries.where(state: :draft).empty?
+  end
+
+  def no_entry_to_balance?
+    journal_entries.where('debit != credit').empty?
+  end
+
+  def unbalanced_radical_account_classes_array
+    # Expected to have a range from 1 to 7
+    radical_numbers = Nomen::Account.items.values.select { |a| a.send(Account.accounting_system)&.match(/^[1-9]$/) }.map { |a| a.send(Account.accounting_system) }
+    balanced_radical_account_classes = []
+    accounts = radical_numbers.map { |rad_numb| Account.where('number ~* ?', '^' + rad_numb + '0*$') }.flatten
+    accounts.each do |account|
+      # Get all journal entry items which is included in one the accounts and in current financial year
+      jei = JournalEntryItem.where(account_id: account.id).where(financial_year_id: self.id)
+      if jei.any? && (jei.sum(:credit) - jei.sum(:debit) == 0)
+        # This array contains parameters to build link_to url
+        balanced_class = {}
+        balanced_class[:id] = account.id
+        balanced_class[:number] = account.number
+        balanced_class[:period] = self.started_on.to_s + '_' + self.stopped_on.to_s
+        balanced_radical_account_classes << balanced_class
+      end
+    end
+    balanced_radical_account_classes
+  end
+
+  def any_invalid_closure_check?
+    checks = [all_previous_financial_years_closed_or_locked?, no_draft_entry?, no_entry_to_balance?, unbalanced_radical_account_classes_array.empty?]
+    checks.any? { |c| c == false } ? true : false
```

Comment:
`(condition) ? true : false` == `condition`
(and `(condition) ? false : true` == `!condition`, basically anytime you write a ternary/if-else whose results are `true` and `false`, you're missing the opportunity to just use the condition as you result.)